portfolio/src/main/webapp/mapScript.js

```diff
@@ -0,0 +1,154 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License. 
+
+$(document).ready(function() {
+    // Enable comment function in main page
+    handleQuantityButton();
+    handleDeleteButton();
+    getComments();
+    // Enable maps
+    initMap();
+});
+
+const getComments = function(max) {
+    fetch('/data?limit=' + max).then(response => response.json()).then((comments) => {
+        const commentsContainer = document.getElementById('comments-container');
+        commentsContainer.innerHTML = ''; 
+        for (let comment of comments) {
+            commentsContainer.appendChild(createCommentElement(comment));
+        }
+    });
+};
+
+/** Creates an <li> element containing text. */
+const createCommentElement = function(comment) {
+    const commentElement = document.createElement('li');
+    commentElement.className = 'comment';
+    commentElement.innerText = comment.content;
+    return commentElement;
+};
+
+const handleQuantityButton = function() {
+    $('#quantity-button').click(function() {
+        let max = $('#quantity').val();
+        getComments(max);
+    });
+};
+
+const deleteAllComments = function() {
+    const request = new Request('/delete-data', {method: 'POST'});
+    fetch(request).then(response => {
+        getComments();
+    });
+};
+
+const handleDeleteButton = function() {
+    $('#delete-button').click(function() {
+        deleteAllComments();
+    });
+};
+
+let map;
```

Comment:
AFAICT, this doesn't need to be global. Can it be local to initMap and passed to any callbacks that need it? I could be wrong, let me know.

Reply:
Haven't fully worked out the hierarchy of functions here, but same for editMarker. Overall, is there a clearer way to organize all these functions?

Reply:
i just tried to make it local, but it seems not work.. i don't know why though

Reply:
Yeah, I tried making it local too and it doesn't work. I *think* that the local variables in `initMap` aren't updated for the next callback to match the values set in the previous callback (but my knowledge of javascript is limited, this was just using the debugger in chrome 🤷‍♀️ )

Reply:
I couldn't find Iris' appspot link so I can't debug :(

I'd still rather the code surrounding the map be a little cleaner and easier to follow (fewer global variables, function flow easier to understand) but it's okay for this purpose